File: Code-Force/Python/Kuriyama.py
import sys

def kuriyama(question, points, x, y):
    if question == 1:
        return sum(points[x-1:y])
    else:
        points.sort()
        return sum(points[x-1:y])

inputs = []
while True:
    line = input("")
    if line == "":
        break
    else :
        inputs.append(line)

pointStr = inputs[1].split(" ")
questionStr = [s.split(" ") for s in inputs[3:]]

points, questions = [int(s) for s in pointStr], [[int(s) for s in innerList] for innerList in questionStr]

results = []

for question in questions:
    result = kuriyama(question[0], points, question[1], question[2])
    results.append(result)

print(results)

File: Code-Force/Haskell/Kuriyama.hs
import Data.List(sort)

main :: IO ()
main = do
    input <- getContents
    let (_:pointStr:_:questionStr) = lines input
        points = map read (words pointStr)
        questions = map (map read . words) questionStr
        sortedPoints = sort points
        results = map (calcPoint points sortedPoints) questions
    mapM_ print results
 
calcPoint :: [Integer] -> [Integer] -> [Integer] -> Integer
calcPoint points sortedPoints [questionType, x, y]
    | questionType == 1 = sum' points (x,y)
    | questionType == 2 = sum' sortedPoints (x,y)
    | otherwise = 0
    where sum' :: [Integer] -> (Integer, Integer) -> Integer
          sum' points (x,y)
            | x > y = 0
            | otherwise = sum $ drop (x' - 1) $ take y' points
          x' = fromInteger x
          y' = fromInteger y